File: packages/react-native-graphics/apple/common/RNGCanvasViewManager.m
#import <React/RCTConvert.h>
#import <React/RCTView.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(RNGCanvasViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(paths, NSArray)
RCT_EXPORT_VIEW_PROPERTY(preserveAspectRatio, NSString)
RCT_EXPORT_VIEW_PROPERTY(viewBox, NSString)

@end

File: packages/react-native-graphics/apple/common/RNGCanvasViewManager.swift
@objc(RNGCanvasViewManager)
class RNGCanvasViewManager: RCTViewManager {
    @objc override static func requiresMainQueueSetup() -> Bool {
        return false
    }

    override func view() -> RCTView? {
        return RNGCanvasView()
    }

    @objc
    override var methodQueue: DispatchQueue {
        get {
            return DispatchQueue.main
        }
    }
}
